internal/service/ecs/cluster.go

```diff
@@ -247,26 +250,26 @@ func resourceClusterRead(d *schema.ResourceData, meta interface{}) error {
 	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
 	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
 
-	var out *ecs.DescribeClustersOutput
+	var cluster *ecs.Cluster
 	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
 		var err error
-		out, err = FindClusterByNameOrARN(conn, d.Id())
+		cluster, err = FindClusterByNameOrARN(context.Background(), conn, d.Id())
 
-		if err != nil {
+		if tfresource.NotFound(err) {
+			if d.IsNewResource() {
+				return resource.RetryableError(err)
+			}
 			return resource.NonRetryableError(err)
 		}
```

Comment:
```suggestion
		if d.IsNewResource() && tfresource.NotFound(err) {
			return resource.RetryableError(err)
		}
```
This is a more provider- and Go-idiomatic way.